lib/usd/translators/lightWriter.cpp

```diff
@@ -13,54 +13,205 @@
+#include <maya/MGlobal.h>
+#include <pxr/usd/usdLux/light.h>
+#include <pxr/usd/usdLux/distantLight.h>
 
 PXR_NAMESPACE_OPEN_SCOPE
 
-PXRUSDMAYA_DEFINE_WRITER(PxrAovLight, args, context)
-{
-    return UsdMayaTranslatorRfMLight::Write(args, context);
-}
+/// directionalLight
+PXRUSDMAYA_REGISTER_WRITER(directionalLight, PxrUsdTranslators_DirectionalLightWriter);
+PXRUSDMAYA_REGISTER_ADAPTOR_SCHEMA(directionalLight, UsdLuxDistantLight);
 
-PXRUSDMAYA_DEFINE_WRITER(PxrCylinderLight, args, context)
+PxrUsdTranslators_DirectionalLightWriter::PxrUsdTranslators_DirectionalLightWriter(
+        const MFnDependencyNode& depNodeFn,
+        const SdfPath&           usdPath,
+        UsdMayaWriteJobContext&  jobCtx)  : UsdMayaPrimWriter(depNodeFn, usdPath, jobCtx)
 {
-    return UsdMayaTranslatorRfMLight::Write(args, context);
+	UsdLuxDistantLight distantLight = UsdLuxDistantLight::Define(GetUsdStage(), GetUsdPath());
+    _usdPrim = distantLight.GetPrim();
+    if (!TF_VERIFY(
+            _usdPrim,
+            "Could not get UsdPrim for UsdLuxDistantLight at path '%s'\n",
+            GetUsdPath().GetText())) {
+        return;
+    }
 }
 
-PXRUSDMAYA_DEFINE_WRITER(PxrDiskLight, args, context)
+/* virtual */
+void PxrUsdTranslators_DirectionalLightWriter::Write(const UsdTimeCode& usdTime)
 {
-    return UsdMayaTranslatorRfMLight::Write(args, context);
+	MStatus status;
+    UsdMayaPrimWriter::Write(usdTime);
+	// Since write() above will take care of any animation on the light's
+    // transform, we only want to proceed here if:
+    // - We are at the default time and NO attributes on the shape are animated.
+    //    OR
+    // - We are at a non-default time and some attribute on the shape IS animated.
+    if (usdTime.IsDefault() == _HasAnimCurves()) {
+        return;
+    }
+    UsdLuxDistantLight primSchema(_usdPrim);
+    MFnDirectionalLight lightFn(GetDagPath(), &status);
+    if (!status) {
+        // Do this just to get the print
+        CHECK_MSTATUS(status);
+        return;
+    }
+    
+    // First write the base light attributes
+	UsdMayaTranslatorLight::WriteLightAttrs(usdTime, primSchema, lightFn, _GetSparseValueWriter());
+    // Then write the specialized attributes for directional lights
+    UsdMayaTranslatorLight::WriteDirectionalLightAttrs(usdTime, primSchema, lightFn, _GetSparseValueWriter());
 }
 
-PXRUSDMAYA_DEFINE_WRITER(PxrDistantLight, args, context)
+/// pointLight
+PXRUSDMAYA_REGISTER_WRITER(pointLight, PxrUsdTranslators_PointLightWriter);
+PXRUSDMAYA_REGISTER_ADAPTOR_SCHEMA(pointLight, UsdLuxSphereLight);
+
+
+PxrUsdTranslators_PointLightWriter::PxrUsdTranslators_PointLightWriter(
+        const MFnDependencyNode& depNodeFn,
+        const SdfPath&           usdPath,
+        UsdMayaWriteJobContext&  jobCtx)  : UsdMayaPrimWriter(depNodeFn, usdPath, jobCtx)
 {
-    return UsdMayaTranslatorRfMLight::Write(args, context);
+    // TODO : do we want to move the creation to fileio/translators ?
```

Comment:
Keeping it separate will allow reusing utilities for prims that are already created in memory. Not a current use-case for us IIRC, but nothing hurts us keeping it in this way.

Reply:
ok, I removed these comments